feat: remove double clicks in favor of extra buttons

luca-schlecker opened the pull request:
Closes #499 

## What I have made

Introduced icon links instead of double clicks.
Hopefully this fixes the e2e-test flakiness.

## Checklist

Either tick or cross out the items that do not apply (using \~\~example text\~\~) and give a reason why the item does not apply.

- [x] I have updated the documentation accordingly and commented my code
- [x] I have added tests that prove my fix is effective or that my feature works
  - [ ] unit tests
  - [x] integration tests
  - [x] end-to-end tests
- [x] (for reviewer) I have checked the implementation against the requirements

mowi12 reviewed (changes requested):
Before I have a closer look at your code, there are two points.

1. Is it possible that the right side (the slide-in window) does not open, if you click on the external-link icon?

Currently, it first opens the side view and the redirects.

https://github.com/user-attachments/assets/ad0c933a-2bfc-4616-9b74-74f8fe8580ad

2. There are 3 teamscale warning. Please check them out and find an appropriate solution.

Generally, on the first look, everything looks pretty good!

luca-schlecker commented:
1. I fixed the issue you've encountered. Click-events for these link icons are now prevented from being propagated.
2. These findings are about empty blocks in the tests. I deem them irrelevant, as these empty blocks are explicitly there.

mowi12 reviewed (changes requested):
Perfect, thank you!
This is the exact behaviour I wanted 👍🏼 

Regarding the teamscale warnings - I agree, this is explicit, but maybe simply add some comments, so this is clear for everyone who looks at the code?

After looking closer at the code, I'm not quite sure about some things.
I checked out `develop` in order to see the old behaviour. I noticed, that there never was any double click behaviour in the reading list, e.g. So my question now is, was this intended? If yes, why didn't the code work, since the `ReadingListEntry` clearly had the code which handled the single/double click code.
And why do we add the new code snippet you added in the three components? I tested in the reading list without the addition, and it still works ...

Could you please test and either confirm or explain why the code is needed if the reading list never displays the external icon?

And maybe it would be easier to create a custom component if we really need to use the code 3 times.

luca-schlecker committed:
feat: add a link button as the preferred alternative to double-clicking

should fix e2e test flakiness

luca-schlecker commented:
Reintroduced untested double-click ability, like discussed in the meeting. Also rebased onto `develop`, reintroducing the phased out tests.

Merseleo commented:
Please turn on the test "When navigating to the project paper view, then the page is displayed" before merging. This test should be fixed with your solution as well and used again.

mowi12 reviewed (changes requested):
This looks good, but please answer to my last review and @Merseleo's request.

luca-schlecker commented:
> Please turn on the test "When navigating to the project paper view, then the page is displayed" before merging. This test should be fixed with your solution as well and used again.

Removed the `fixme` from this test :+1:, thank you for the hint.

luca-schlecker committed:
fix: reenable test

luca-schlecker committed:
docs: comment empty callbacks

luca-schlecker commented:
> Perfect, thank you! This is the exact behaviour I wanted 👍🏼
> 
> Regarding the teamscale warnings - I agree, this is explicit, but maybe simply add some comments, so this is clear for everyone who looks at the code?
> 
> After looking closer at the code, I'm not quite sure about some things. I checked out `develop` in order to see the old behaviour. I noticed, that there never was any double click behaviour in the reading list, e.g. So my question now is, was this intended? If yes, why didn't the code work, since the `ReadingListEntry` clearly had the code which handled the single/double click code. And why do we add the new code snippet you added in the three components? I tested in the reading list without the addition, and it still works ...
> 
> Could you please test and either confirm or explain why the code is needed if the reading list never displays the external icon?
> 
> And maybe it would be easier to create a custom component if we really need to use the code 3 times.

I added comments to these occasions. The reading-list entry was solely adjusted for consistency's sake. I wondered about this too, but I deemed it to be something irrelevant for this PR. Furthermore, I repeated myself because it is a small, limited amount and the other two probably will be obsolete anyway. There was never an issue with the other components, I adjusted it just to be consistent. It stems from the original idea to remove double-click entirely. These components don't even use it as far as I am aware.

luca-schlecker committed:
style: format code

mowi12 reviewed (changes requested):
This looks good, thanks!

I only have one last question.
I saw, that you added `gap-2` in only one of the modified list entry components. Have you tested, how the other list entries look, if a custom `onclick` function is provided?

Don't we need to modify the surrounding div/component as well, or is this already handled by the current CSS?

luca-schlecker commented:
> This looks good, thanks!
> 
> I only have one last question. I saw, that you added `gap-2` in only one of the modified list entry components. Have you tested, how the other list entries look, if a custom `onclick` function is provided?
> 
> Don't we need to modify the surrounding div/component as well, or is this already handled by the current CSS?

I did not test the others with the link icon shown. As that functionality is currently not used in the frontend, what would you suggest? Test it now, leave it until needed, or just leave it as is as these parts could likely be refactored in the future?

mowi12 reviewed (commented):
I think the best way to continue is to simply test this by testing this now.
If the `gap-2` should be there, I think we should add it now, as we are already aware of it.

luca-schlecker committed:
fix: link button presentation

luca-schlecker commented:
I now manually verified that all three of these entry components share the same look. Integration tests were already in place.

mowi12 reviewed (dismissed):
Great, then lets finish this!

Only some formatting is left : )

luca-schlecker commented:
> Great, then lets finish this!
> 
> Only some formatting is left : )

Done, also rebased.
